addons/payment_mollie/models/payment_transaction.py

```diff
@@ -76,6 +77,33 @@ def _mollie_prepare_payment_request_payload(self):
             # redirection, we include it in the redirect URL to be able to match the transaction.
             'redirectUrl': f'{redirect_url}?ref={self.reference}',
             'webhookUrl': f'{webhook_url}?ref={self.reference}',
+            'billingAddress': self._prepare_billing_address(),
+            'lines': [{
+                "description": 'Odoo purchase',
+                "quantity": 1,
+                "unitPrice": {
+                    "currency": self.currency_id.name,
+                    "value": f"{self.amount:.{decimal_places}f}"
+                },
+                "totalAmount": {
+                    "currency": self.currency_id.name,
+                    "value": f"{self.amount:.{decimal_places}f}"
+                },
+            }],
+        }
+
+        return payload
+
+    def _prepare_billing_address(self):
+        name_parts = payment_utils.split_partner_name(self.partner_name)
```

Comment:
You can deconstruct it like this, then it will be easier to check for in the dict
```suggestion
        first_name, last_name = payment_utils.split_partner_name(self.partner_name)
```